sys/socket.c

```diff
@@ -435,3 +436,120 @@ int socketpair(int domain, int type, int protocol, int socket_vector[2])
 
 	return 0;
 }
+
+
+int getifaddrs(struct ifaddrs **ifap)
+{
+	struct ifaddrs *ifa;
+	msg_t msg = { 0 };
+	size_t bufsz, namesz, addrsz;
+	char *p;
+	sockport_resp_t *smo = (sockport_resp_t *)msg.o.raw;
+
+	msg.type = sockmGetIfAddrs;
+	msg.o.size = 256;
+
+	for (;;) {
+		msg.o.data = realloc(msg.o.data, msg.o.size);
+		if (!msg.o.data) {
+			free(msg.o.data);
+			return EAI_MEMORY;
+		}
```

Comment:
```suggestion
		void *tmp;
		tmp = realloc(msg.o.data, msg.o.size);
		if (tmp == NULL) {
			free(msg.o.data);
			return EAI_MEMORY;
		}
		msg.o.data = tmp;
		
```
Memory leak.
If `realloc()` returns `NULL` you will lose previous value of `msg.o.data`.

Reply:
You're right here. I'll fix that.